lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/MultiRuntimeYarnBootstrapDriverConfigGenerator.java

```diff
@@ -0,0 +1,310 @@
+
+  private final REEFFileNames reefFileNames;
+  private final ConfigurationSerializer configurationSerializer;
+
+  @Inject
+  private MultiRuntimeYarnBootstrapDriverConfigGenerator(final REEFFileNames reefFileNames,
+                                                         final ConfigurationSerializer configurationSerializer) {
+    this.configurationSerializer = configurationSerializer;
+    this.reefFileNames = reefFileNames;
+  }
+
+  /**
+   * Adds yarn runtime definitions to the builder.
+   * @param yarnJobSubmissionParams Yarn job submission parameters
+   * @param jobSubmissionParameters Generic job submission parameters
+   * @param builder The multi runtime builder
+   */
+  private void addYarnRuntimeDefinition(
+          final AvroYarnJobSubmissionParameters yarnJobSubmissionParams,
+          final AvroJobSubmissionParameters jobSubmissionParameters,
+          final MultiRuntimeDefinitionBuilder builder) {
+    // create and serialize yarn configuration if defined
+    final Configuration yarnDriverConfiguration =
+            createYarnConfiguration(yarnJobSubmissionParams, jobSubmissionParameters);
+
+    // add yarn runtime to the builder
+    builder.addRuntime(yarnDriverConfiguration, RuntimeIdentifier.RUNTIME_NAME);
+  }
+
+  private Configuration createYarnConfiguration(
+          final AvroYarnJobSubmissionParameters yarnJobSubmissionParams,
+          final AvroJobSubmissionParameters jobSubmissionParameters) {
+    return YarnDriverConfiguration.CONF
+            .set(YarnDriverConfiguration.JOB_SUBMISSION_DIRECTORY,
+                    yarnJobSubmissionParams.getDfsJobSubmissionFolder().toString())
+            .set(YarnDriverConfiguration.JOB_IDENTIFIER,
+                    jobSubmissionParameters.getJobId().toString())
+            .set(YarnDriverConfiguration.CLIENT_REMOTE_IDENTIFIER,
+                    ClientRemoteIdentifier.NONE)
+            .set(YarnDriverConfiguration.JVM_HEAP_SLACK, 0.0)
+            .set(YarnDriverConfiguration.RUNTIME_NAMES, RuntimeIdentifier.RUNTIME_NAME)
+            .build();
+  }
+
+  /**
+   * Adds yarn runtime definitions to the builder, with a dumy name, to prevent evaluator submission.
+   * @param yarnJobSubmissionParams Yarn job submission parameters
+   * @param jobSubmissionParameters Generic job submission parameters
+   * @param builder The multi runtime builder
+   */
+  private void addDummyYarnRuntimeDefinition(
+          final AvroYarnJobSubmissionParameters yarnJobSubmissionParams,
+          final AvroJobSubmissionParameters jobSubmissionParameters,
+          final MultiRuntimeDefinitionBuilder builder) {
+    // create and serialize yarn configuration if defined
+    final Configuration yarnDriverConfiguration =
+            createYarnConfiguration(yarnJobSubmissionParams, jobSubmissionParameters);
+    // add yarn runtime to the builder
+    builder.addRuntime(yarnDriverConfiguration, DUMMY_YARN_RUNTIME);
```

Comment:
Can you remind me what submission looks like? How does this prevent submission to the YARN environment?